### ISSUE TITLE
Consider using ProseMirror for RichTextInput?

### ISSUE THREAD

**Issue by kleinerpirat** [ankitects/anki#2108]
While doing research for this forum topic, I discovered [ProseMirror](https://forums.ankiweb.net/t/using-markdown-to-highlight-text/23621/5) - a rich-text editor toolkit made by the same guy that wrote CodeMirror.

It has several very intriguing features that would fit well into Anki:
- WYSIWYM Markdown editing (https://prosemirror.net/examples/markdown/)
- Node views (https://prosemirror.net/docs/ref/#view.NodeView)
  - allows inline CodeMirror instances (https://prosemirror.net/examples/codemirror/)
  - together with [this plugin](https://benrbray.com/prosemirror-math/), we could replace the buggy MathJax editor with inline KaTex (#2106)

**Comment by dae** [comment]
The comments on https://github.com/ankitects/anki/issues/2025 apply here as well. 

One other concern is that CodeMirror accounts for something like half the editor load time at the moment when there are lots of fields, so we'd want to check how it performs on larger notetypes.

**Comment by kvn1351** [comment]
I still think that TinyMCE is the best choice. I've even had some success making it work but had to stop playing around with it for the time being.

**Comment by FernandoBasso** [comment]
We definitely need something that does some nice markup language. In that regard, nothing compares to Asciidoctor, but Markdown is the second best option as far as markup goes.

Asciidoctor is as simple as Markdown for the simple things, and has room for many more rich features if required/desirable.

From time to time people write a Markdown plugin that stops working after some Anki update, or just gets abandoned.

The default HMTL generated by Anki is not a very good one, and I think many people would love a nice markup language like Asciidoctor or Markdown supported natively.

**Comment by navakelvin** [comment]
> The default HMTL generated by Anki is not a very good one, and I think many people would love a nice markup language like Asciidoctor or Markdown supported natively.

Markdown would be great. It is by far more popular.

**Comment by FernandoBasso** [comment]
> Markdown would be great. It is by far more popular.

Yes, more popular, but not better than Asciidoctor (in terms of features). Asciidoctor lets you go easy (if you want), with a very minimal set of markup syntax (title, headings, lists, formatting code), but also allows for much (and I mean **much**) more if needed.

That said, I would be glad with Markdown too :)
